app/uk/gov/hmrc/channelpreferences/controllers/PreferenceController.scala

```diff
@@ -76,4 +83,103 @@ class PreferenceController @Inject()(
+             |   And an error will be generated
+             */
+            reply(UNAUTHORIZED, s"entityId already has a different itsaId linked to it in entity resolver")
+
+            // TODO Isn't the following acceptance criteria already included in Case 1.5 ???
+            /*
+             | Case 1.6 - itsaId is already linked to a different entityId in entity resolver.
+             |
+             |   Given I am a customer who has successfully enrolled in ITSA
+             |   When my itsaId is already linked to a different entityId in entity resolver
+             |   Then my itsaId will not be added (i.e linked to) any entityId
+             |   And an error will be generated
+           */
+          }
+        } else {
+          // entity.itsa.nonDefined, which means no link has been created yet
+          if (!authToken_saUtr.isDefined) {
+            /*
+             | Case 1.4 - SAUTR does not exist in the  Auth token
+             |
+             |   Given I am a customer who has successfully enrolled in ITSA
+             |   When SAUTR does not exist in the Auth token
+             |   Then my itsaId will be added (i.e linked to) the entityId
+             */
+            entityResolverConnector
+              .update(resolvedEntity.copy(itsa = Some(passedBack_itsaId)))
+              .flatMap { _ =>
+                reply(OK, "itsaId successfully linked to entityId")
+              }
+          } else {
+            // authToken_saUtr.isDefined
+            if (resolvedEntity.saUtr.isDefined && (resolvedEntity.saUtr.get == authToken_saUtr.get)) {
+              /*
+               | Case 1.1 - SAUTR in Auth token is the same as SAUTR in entity resolver
+               |
+               |   Given I am a customer who has successfully enrolled in ITSA
+               |   When SAUTR in Auth token is the same as SAUTR in entity resolver
+               |   Then my itsaId will be added (i.e linked to) the entityId
+               */
+              entityResolverConnector
+                .update(resolvedEntity.copy(itsa = Some(passedBack_itsaId)))
+                .flatMap { _ =>
+                  reply(OK, "itsaId successfully linked to entityId")
+                }
+            } else {
+              /*
+               | Case 1.2 - SAUTR in Auth token is different from  SAUTR in entity resolver
+               |
+               |   Given I am a customer who has successfully enrolled in ITSA
+               |   When SAUTR in Auth token is different from  SAUTR in entity resolver
+               |   Then my itsaId will not be added (i.e linked to) any entityId
+               |   And an error will be generated
+               */
+              reply(UNAUTHORIZED, "SAUTR in Auth token is different from SAUTR in entity resolver")
+
+              // TODO Isn't the following acceptance criteria already included in Case 1.2 ???
+              // Case 1.3 - SAUTR in Auth token is linked to a different entityId  in entity resolver
```

Comment:
My understanding (might be worth confirming with @satyajosyula2020 or Micheal), but for case 1.3, you would have to get the entity linked to the SaUTR from the auth token, and compare it to `resolvedEntity` (i.e the entity got from the `passedBack_entityId`) and see whether the entity Id of those 2 entities matches

Reply:
Thanks for you explanation, @valydia 
I will ask @satyajosyula2020 and Michael to confirm it.